src/ServerRequestFactory.php

```diff
@@ -198,21 +198,18 @@ public static function marshalHeaders(array $server)
     {
         $headers = [];
         foreach ($server as $key => $value) {
-            if ($value && strpos($key, 'HTTP_') === 0) {
-                $name = strtr(substr($key, 5), '_', ' ');
-                $name = strtr(ucwords(strtolower($name)), ' ', '-');
+            if ($value) {
+                if (strpos($key, 'HTTP_') === 0) {
+                    $name = substr($key, 5);
+                } elseif (strpos($key, 'CONTENT_') === 0) {
+                    $name = $key;
+                } else {
+                    continue;
+                }
+
+                $name = strtr($name, '_', '-');
                 $name = strtolower($name);
-
                 $headers[$name] = $value;
-                continue;
-            }
-
-            if ($value && strpos($key, 'CONTENT_') === 0) {
-                $name = substr($key, 8); // Content-
-                $name = 'Content-' . (($name == 'MD5') ? $name : ucfirst(strtolower($name)));
```

Comment:
If you analyze it correctly - it basically does nothing.
First we cut `Content-` at the beginning, then add if unconditionally (why?). After this we check whether remainder is equal to `MD5` and if so - pass it directly, otherwise we make reminder lowercase and than make first character uppercase. After all manipulations we make it lowercase (which basically neglects everything from previous sentence).
So:
* cutting `CONTENT_` and replacing with `Content-` seems reasonable, but can be just replaced with `strtr()` (the real difference in practice is just one symbol) in order to keep is homogeneous with previous block of code
* all the game with repeated upper/lowercasing doesn't make any sense

As far as I saw there are test cases for all situations touched by these changes.